sale_timesheet_timeline/views/sale_order_views.xml

```diff
@@ -0,0 +1,98 @@
+                <field name="task_user_ids" />
+                <field name="product_uom_qty" />
+                <templates>
+                    <t t-name="timeline-item">
+                        <div class="o_project_timeline_item">
+                            <t t-foreach="record.task_user_ids" t-as="user">
+                                <img
+                                    t-if="record.task_user_ids"
+                                    t-attf-src="/web/image/res.users/#{user}/avatar_128/16x16"
+                                    t-att-title="record.user"
+                                    width="16"
+                                    height="16"
+                                    class="mr8"
+                                    alt="User"
+                                />
+                            </t>
+                            <span name="name">
+                                <t t-esc="record.name.split('\n')[0]" />
+                            </span>
+                            <small
+                                name="planned_hours"
+                                class="text-info ml4"
+                                t-if="record.product_uom_qty"
+                            >
+                                <t
+                                    t-esc="field_utils.format.float_time(record.product_uom_qty)"
+                                />
+                            </small>
+                        </div>
+                    </t>
+                </templates>
+            </timeline>
+        </field>
+    </record>
+    <record id="action_sale_order_line_timeline" model="ir.actions.act_window">
+        <field name="name">Sales Tasks Planning</field>
+        <field name="res_model">sale.order.line</field>
+        <field name="view_mode">timeline</field>
+        <field
+            name="domain"
+        >[('order_id', '=', active_id), ('product_type', '=', 'service')]</field>
+    </record>
+    <!-- Put this one after the action definition -->
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree" position="inside">
+                <field
+                    name="task_date_start"
+                    optional="hide"
+                    attrs="{'invisible': [('product_type', '!=', 'service')]}"
+                />
+                <field
+                    name="task_date_end"
+                    optional="hide"
+                    attrs="{'invisible': [('product_type', '!=', 'service')]}"
+                />
+                <field
+                    name="task_user_ids"
+                    optional="hide"
+                    widget="many2many_avatar_user"
+                    attrs="{'invisible': [('product_type', '!=', 'service')]}"
+                />
```

Comment:
If we can edit a record when the sale order is confirmed, then these changes must be propagated to the task. Otherwise, we should block edits when the sale order is confirmed. What do you think?

![image](https://github.com/user-attachments/assets/5fb43e83-57c3-41ae-a46b-e15633463639)

Reply:
what about this?

Reply:
ping @pedrobaeza

Reply:
@carlos-lopez-tecnativa @pilarvargas-tecnativa fixed